leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -7,61 +8,404 @@
+                    funcSegList[0].set('refEvidencePair', initValue); // Set first index of funcSegList to given initValue 
+            }
+            this.get('evolvingFunction').set('type', 'NT');
+            this.get('evolvingFunction').set('functionSegList', []);
+        }     
+    }, 
+ 
+    /**
+     * Sets the initial satisfaction value for this Intention to '(no value)'
+     */
+    removeInitialSatValue: function() {
+
+        this.changeInitialSatValue('(no value)');
+    },    
+ 
+    /**
+     * Returns the 4 digit representation for the
+     * Intention's initial satisfaction value
+     *
+     * @returns {String}
+     */
+    getInitialSatValue: function() {
+
+        var intentionEval = graph.getUserEvaluationBBM(this.cid, '0'); // Finding UserEvaluationBBM with same ID and startTP as this IntentionBBM 
+        if (typeof intentionEval == 'undefined'){
+            return '(no value)';
+        } else {
+            return intentionEval.get('assignedEvidencePair'); 
+        }
+    }, 
+ 
+    /**
+     * Resets evolving functions
+     */
+    removeFunction: function() {
+
+        this.removeAbsConstraint();
+        this.model.get('intention').set('evolvingFunction', null); // Set to null 
+    }, 
+ 
+    /**
+     * Clears all FuncSegments for the Intention's
+     * EvolvingFunction and adds new FuncSegments according to the current
+     * function type.
+     * Called when user changes the function type in Element Inspector 
+     * @param {String} funcType
+     */
+    setEvolvingFunction: function(funcType) {
+
+        this.set('evolvingFunction', new EvolvingFunctionBBM({type: funcType, functionSegList: []}));
+
+        // Since function changed, remove all current absolute constraints related to this intention
+        this.removeAbsConstraint();
+ 
+        // Add new absolute constraints if required
+        this.addAbsConstraint(funcType);
+ 
+        // var initValue = graph.get('userEvaluationList').get(this.cid, '0').get('assignedEvidencePair');
+        var initValue = graph.getUserEvaluationBBM(this.cid, '0').get('assignedEvidencePair');
+ 
+
+        // Creates the correct FunctionSegmentBBM(s) for the selected function type
+        if (funcType == 'C' || funcType == 'R' || funcType == 'I' || funcType == 'D' || funcType == 'UD') {
+            if (funcType == 'C') {
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: initValue, startTP: '0', startAT: 0});  
+            } else if (funcType == 'R') {
+                // The reference evidence pair for a Stochastic function is always 0000
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: '(no value)', startTP: '0', startAT: 0}); 
+            } else if (funcType == 'I' || funcType == 'D') {
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: null, startTP: '0', startAT: 0}); 
```

Comment:
Why is the refEvidencePair null for this one?

Reply:
It was defined this way previously, @amgrubb do you know if this is correct?

Reply:
We are changing these defaults to fully satisfied for increasing and fully denied for decreasing